shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-provider/shardingsphere-db-discovery-mgr/src/main/java/org/apache/shardingsphere/dbdiscovery/mgr/MGRDatabaseDiscoveryType.java

```diff
@@ -114,6 +115,31 @@ private void checkIsSinglePrimaryMode(final Statement statement) throws SQLExcep
         }
     }
     
+    private void checkDataSourceInReplicationGroup(final Statement statement, final Map<String, DataSource> dataSourceMap) throws SQLException {
+        List<String> memberDataSourceURLs = new LinkedList<>();
+        try (ResultSet resultSet = statement.executeQuery(MEMBER_LIST)) {
+            while (resultSet.next()) {
+                memberDataSourceURLs.add(String.format("%s:%s", resultSet.getString("MEMBER_HOST"), resultSet.getString("MEMBER_PORT")));
+            }
+        }
+        checkDataSourcesExistGroupMember(dataSourceMap, memberDataSourceURLs);
+    }
+    
+    private void checkDataSourcesExistGroupMember(final Map<String, DataSource> dataSourceMap, final List<String> memberDataSourceURLs) throws SQLException {
```

Comment:
`List<String> memberDataSourceURLs` can be `Collection<String> memberDataSourceURLs`